components/roadmap-grid/RoadmapDetailedView.tsx

```diff
@@ -145,28 +147,38 @@ export function RoadmapDetailed({
   return (
     <>
       {isDevMode && <NumSlider msg="how many header ticks" value={numHeaderTicks} min={5} max={60} setValue={setNumHeaderTicks}/>}
-      {isDevMode && <NumSlider msg="how many grid columns" value={numGridCols} min={20} max={60} step={numHeaderTicks} setValue={setNumGridCols}/>}
-
-      <Box className={`${styles.timelineBox} ${ viewMode=='detail' ? styles.detailView : '' }`} >
-        <Grid ticksLength={numGridCols}>
-          {ticksHeader.map((tick, index) => (
-
-            <GridHeader key={index} tick={tick} index={index} numHeaderTicks={numHeaderTicks} numGridCols={numGridCols}/>
-          ))}
+      {isDevMode && <NumSlider msg="how many grid columns" value={numGridCols} min={20} max={60} step={numHeaderTicks} setValue={setNumGridCols} />}
+
+      {globalLoadingState.get() ? (
+        <Stack pt={"20px"} hidden={!globalLoadingState.get()}>
+          <Skeleton isLoaded={!globalLoadingState.get()} height='60px' />
+          <Skeleton isLoaded={!globalLoadingState.get()} height='150px' />
+          <Skeleton isLoaded={!globalLoadingState.get()} height='150px' />
+          <Skeleton isLoaded={!globalLoadingState.get()} height='150px' />
+          <Skeleton isLoaded={!globalLoadingState.get()} height='150px' />
+        </Stack>
+      ) : (
+        <Box className={`${styles.timelineBox} ${viewMode == 'detail' ? styles.detailView : ''}`} >
+          <Grid ticksLength={numGridCols}>
+            {ticksHeader.map((tick, index) => (
+
+              <GridHeader key={index} tick={tick} index={index} numHeaderTicks={numHeaderTicks} numGridCols={numGridCols} />
+            ))}
 
-          <Headerline numGridCols={numGridCols} ticksRatio={3}/>
-        </Grid>
-        <Grid ticksLength={numGridCols} scroll={true}  renderTodayLine={showTodayMarker} >
-          {issuesGroupedState.map((group, index) => (
+            <Headerline numGridCols={numGridCols} ticksRatio={3} />
+          </Grid>
+          <Grid ticksLength={numGridCols} scroll={true} renderTodayLine={showTodayMarker} >
+            {issuesGroupedState.map((group, index) => (
               <ErrorBoundary key={`Fragment-${index}`} >
-                <GroupHeader group={group} key={`GroupHeader-${index}`} issueDataState={issueDataState}/><GroupWrapper key={`GroupWrapper-${index}`}>
+                <GroupHeader group={group} key={`GroupHeader-${index}`} issueDataState={issueDataState} /><GroupWrapper key={`GroupWrapper-${index}`}>
                   {group.ornull != null && group.items.ornull != null &&
                     _.sortBy(group.items.ornull, ['title']).map((item, index) => <GridRow key={index} milestone={item} index={index} timelineTicks={ticks} numGridCols={numGridCols} numHeaderItems={numHeaderTicks} issueDataState={issueDataState} />)}
                 </GroupWrapper>
               </ErrorBoundary>
             ))}
-        </Grid>
-      </Box>
+          </Grid>
+        </Box>
```

Comment:
We should probably move the skeletons around each component instead of obfuscating above, but I'm not clear on the errors you were experiencing.